frontends/web/src/routes/buy/info.tsx

```diff
@@ -83,11 +90,8 @@ class BuyInfo extends Component<Props, State> {
         // @ts-ignore
         .then(accounts => accounts.map(({ name, code }) => ({ text: name, value: code })))
         .then(options => {
-            if (this.state.status === 'choose' && options.length === 1) {
-                route(`/buy/moonpay/${options[0].value}`);
-            } else {
-                this.setState({ options });
-            }
+            this.setState({ options });
+            this.maybeProceed();
```

Comment:
It is unclear when preact updates the state after `this.setState({ options })`, use the 2nd argument of setState that is a callback when the state has updated: `this.setState({ options }, this.maybeProceed);`
or 
```
this.setState({ options }, () => this.maybeProceed());
```

Reply:
Good catch. The preact docs suck though, there is nothing about setState? https://preactjs.com/guide/v10/components

Reply:
- https://preactjs.com/guide/v10/upgrade-guide/#dont-access-thisstate-synchronously
- https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous